src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -392,11 +397,11 @@ const EnvironmentPicker = ({
 
     const containerDisplayText = useMemo(() => {
         const displayText = getContainerDisplayText(
-            getContainerNameFromUrl(storage.containerUrl),
-            getStorageAccountUrlFromContainerUrl(storage.containerUrl)
+            getContainerNameFromUrl(storage?.containerUrl),
+            getStorageAccountUrlFromContainerUrl(storage?.containerUrl)
         );
         return displayText || t('environmentPicker.noContainer');
-    }, [t, storage.containerUrl]);
+    }, [t, storage?.containerUrl]);
```

Comment:
how safe is this? I don't think I've used it before.

Reply:
yep it is possible, React keeps track of changes in the dependency array, storage?.containerUrl can be either `undefined` or a value. [undefined] is same as []. So in either way it treats it as a change and call useEffect again: https://stackoverflow.com/questions/61439712/react-conditional-dependency-on-useeffect-or-how-to-prevent-redundancy